c/example04.c

```diff
@@ -0,0 +1,154 @@
+        return NULL;
+    }
+
+    printf("\nTag created successfully:\n");
+    printf("Resources ID: %s\n", tag->_resources_id);
+    printf("Key: %s\n", tag->_key);
+    printf("Value: %s\n", tag->_value);
+
+    return tag;
+}
+
+/* Function to delete a tag */
+int delete_tag(struct Tag *tag)
+{
+    CURLcode res;
+
+    char json_data[BUFSIZ];
+    snprintf(json_data, sizeof(json_data), "{\"ResourceIds\": [\"%s\"], \"Tags\": [{\"Key\": \"%s\", \"Value\": \"%s\"}]}", tag->_resources_id, tag->_key, tag->_value);
+
+    /* Setting the URL for delete */
+    curl_easy_setopt(tag->curl, CURLOPT_URL, "https://api.eu-west-2.outscale.com/api/v1/DeleteTags");
+
+    /* Setting the AWS SigV4 and AK/SK */
+    curl_easy_setopt(tag->curl, CURLOPT_AWS_SIGV4, "osc");
+    curl_easy_setopt(tag->curl, CURLOPT_USERNAME, tag->_ak);
+    curl_easy_setopt(tag->curl, CURLOPT_PASSWORD, tag->_sk);
+
+    /* Empty post field to indicate we want to send a POST request */
+    curl_easy_setopt(tag->curl, CURLOPT_POSTFIELDS, json_data);
+
+    /* Set the Content-Type header to specify JSON */
+    struct curl_slist *headers = NULL;
+    headers = curl_slist_append(headers, "Content-Type: application/json");
+    curl_easy_setopt(tag->curl, CURLOPT_HTTPHEADER, headers);
+
+    /* Empty post field to indicate we want to send a POST request */
+    curl_easy_setopt(tag->curl, CURLOPT_POSTFIELDS, json_data);
+
+    res = curl_easy_perform(tag->curl);
+
+    if (res != CURLE_OK) {
+        fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
+        return 1;
+    }
+
+    printf("\nTag deleted successfully:\n");
+    printf("Resources ID: %s\n", tag->_resources_id);
+    printf("Key: %s\n", tag->_key);
+    printf("Value: %s\n", tag->_value);
+
+    return 0;
+}
+
+int main(int argc, char *argv[])
+{
+    /* Getting the access key / secret key from environment */
+    const char *ak = getenv("OSC_ACCESS_KEY");
+    const char *sk = getenv("OSC_SECRET_KEY");
+
+    /* Error handling */
+    if (ak == NULL || sk == NULL) {
+        fprintf(stderr, "OSC_ACCESS_KEY or OSC_SECRET_KEY environment variable not set\n");
+        return 1;
+    }
+
+    /* Error handling */
+    if (strlen(ak) != AK_SIZE || strlen(sk) != SK_SIZE) {
+        fprintf(stderr, "Wrong size OSC_ACCESS_KEY or OSC_SECRET_KEY\n");
+        return 1;
+    }
+
+    /* Error handling */
+    if (argc != 4) {
+        fprintf(stderr, "Usage: ./example04 <ResourceId> <Key> <Value>\n");
+        return EXIT_FAILURE;
+    }
+
+    /* Creating the handler */
+    CURL *curl = curl_easy_init();
+
+    /* Error handling */
+    if (!curl) {
+        fprintf(stderr, "Failed to initialize libcurl.\n");
+        return 1;
+    }
+
+    struct Tag *newTag = create_tag(&(struct Tag){
+        ._resources_id = argv[1],
+        ._key = argv[2],
+        ._value = argv[3],
+        ._ak = ak,
+        ._sk = sk,
+        .curl = curl
+    });
+
+    if (newTag != NULL)
```

Comment:
If you use cleanup, you don't need to call this, at all.
http://echorand.me/site/notes/articles/c_cleanup/cleanup_attribute_c.html
something in the line of:
```C
 void delete_tag_cleanup(struct Tag **newTag) {
   if (*newTag)
      delete_tag(newTag);
}

#define auto_tag  __attribute__ ((__cleanup__(delete_tag_cleanup)));
```

Reply:
THANKS